pkg/specgenutil/volumes.go

```diff
@@ -579,7 +587,21 @@ func getNamedVolume(args []string) (*specgen.NamedVolume, error) {
 	if len(mnt.Destination) == 0 {
 		return nil, errNoDest
 	}
-	newVolume.Options = mnt.Options
+
+	safeOpts := make([]string, 0, len(mnt.Options))
+	for _, opt := range mnt.Options {
+		splitOpt := strings.SplitN(opt, "=", 2)
+		if splitOpt[0] == "subpath" {
+			if len(splitOpt) != 2 {
+				return nil, fmt.Errorf("subpath was not provided a path")
+			}
+			newVolume.SubPath = splitOpt[1]
+		} else {
+			safeOpts = append(safeOpts, opt)
+		}
+	}
```

Comment:
Please not another "random" place to parse options.  Can't we just fix parseMountOptions() to return  a suitable type? We already have to many different places where we pass mount options

Reply:
Done